### PR TITLE
TryRuby - Lesson (Section2)

### PR DESCRIPTION
**Pull request by philipd97**
- Practice on JSON format Hash
- initializing Class
- using of Class method:
-- initialize
-- attr_reader
-- attr_writer
-- attr_accessor

### CODE REVIEW

**Review by tlkiong — CHANGES_REQUESTED** [review]
Can you have a consistent formatting of your code?
If your bracket is on same line, put all same line. if on new line, put all new line

File: section2/try_ruby_lesson.rb [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+	  "3"=>  {"title"=>"Henry VI, Part 2", "finished"=>1591}, 
+	  "4"=>  {"title"=>"Henry VI, Part 3", "finished"=>1591}, 
+	  "5"=>  {"title"=>"Henry VI, Part 1", "finished"=>1592}, 
+	  "6"=>  {"title"=>"Titus Andronicus", "finished"=>1592}, 
+	  "7"=>  {"title"=>"Richard III", "finished"=>1593}, 
+	  "8"=>  {"title"=>"Edward III", "finished"=>1593}, 
+	  "9"=>  {"title"=>"The Comedy of Errors", "finished"=>1594},
+	  "10"=> {"title"=>"Love's Labour's Lost", "finished"=>1595}
+    }
+  }
```
Comment (philipd97) [inline section2/try_ruby_lesson.rb]: @tlkiong do you mean for this hash, put it into one line? If I wanted the one line format like others?
eg: `.select { |key,value| value["finished"] == year }`

Comment (tlkiong) [reply section2/try_ruby_lesson.rb]: no. i am asking you to have **consistency** in your code
image

**Review by tlkiong — CHANGES_REQUESTED** [review]
image
This is to close what?
Keep things consistent but also in a way that its readable.
Your current way is not readable at all.

Try reading your own code

**Comment by philipd97** [comment]
> image
> This is to close what?
> Keep things consistent but also in a way that its readable.
> Your current way is not readable at all.
> 
> Try reading your own code

Hmm.. is it ok if I do it like this?

image

**Comment by tlkiong** [comment]
So, William Shakespeare and 1,2,3,4,5,6,7,8,9,10 all on the same level?

**Comment by philipd97** [comment]
William Shakespeare is one level above 1,2,3,4,5,6,7,8,9,10.
Or should I do it like this? 😅  
 
image

**Comment by tlkiong** [comment]
ask RJ see what he says

**Comment by philipd97** [comment]
@LeeRenJie 👀 what do you think between two of these format? Or is there any format I can refer?

**(1)**
> image

**(2)**
> image

**Comment by LeeRenJie** [comment]
> @LeeRenJie 👀 what do you think between two of these format? Or is there any format I can refer?
> 
> **(1)**
> 
> > image
> 
> **(2)**
> 
> > image

@philipd97  @tlkiong Not sure if this is the right style but this is how I would do it. Maybe looking at the [style guide](https://github.com/rubocop/ruby-style-guide) might help

```ruby
def get_shakey
  { 
    "William Shakespeare" => {
      "1" => { "title" => "The Two Gentlemen of Verona", "finished" => 1591 }, 
      "2" => { "title" => "The Taming of the Shrew", "finished" => 1591 }, 
      "3" => { "title" => "Henry VI, Part 2", "finished" => 1591 }, 
      "4" => { "title" => "Henry VI, Part 3", "finished" => 1591 }, 
      "5" => { "title" => "Henry VI, Part 1", "finished" => 1592 }, 
      "6" => { "title" => "Titus Andronicus", "finished" => 1592 }, 
      "7" => { "title" => "Richard III", "finished" => 1593 }, 
      "8" => { "title" => "Edward III", "finished" => 1593 }, 
      "9" => { "title" => "The Comedy of Errors", "finished" => 1594 },
      "10" => { "title" => "Love's Labour's Lost", "finished" => 1595 }
    }
  }
end

**Review by tlkiong — CHANGES_REQUESTED** [review]
??

**Comment by philipd97** [comment]
> ??

I refactor the Hash bracket, I feel like the readability is better this way too. Is it ok?

**Comment by tlkiong** [comment]
that `??` is asking, what refactoring you did?
image

**Comment by tlkiong** [comment]
Based on your last refactor, this is what you have:
image
Look at it. Doesn't it show that William Shakespeare,1,2,3,4,5,6,7,8,9,10 are all the same level?

---

image
This is the right one. We can clearly distinguish the level. The one you did is completely 1000% wrong.
Did you read the style guide or google how others do it?

**Comment by philipd97** [comment]
Ok I got it now, I missed out on the the hint you mention on the same level & detail on RJ's answer. My bad. I will take note on it clearly next time.

> Did you read the style guide or google how others do it?

In style guide I found only about bracket spacing. Yeah, but I missed out on the details of same level hierarchy.

**Review by tlkiong — APPROVED** [review]

**Review by LeeRenJie — APPROVED** [review]

**Pull request merged**